### PR TITLE
app-office/sc-im: add 0.8.3

### PR DESCRIPTION
**Pull request by ephemer4l**

### CODE REVIEW

**Review by juippis — COMMENTED** [review]
With `+X` it fails to
```
checking for X11/Xmu/Atoms.h... no
configure: error: *** X11/Xmu/Atoms.h is missing ***
```

File: app-office/sc-im/sc-im-0.8.3.ebuild [hunk 1]
```diff
@@ -0,0 +1,142 @@
+# Copyright 2022-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+LUA_COMPAT=( lua5-1 luajit )
```
[the remark this answers is not in the document]
Comment (ephemer4l) [reply app-office/sc-im/sc-im-0.8.3.ebuild:6]: On the installation guide of sc-im they ask for lua 5.1 but I'll find out if anything newer is compatible.

File: app-office/sc-im/sc-im-0.8.3.ebuild [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+DESCRIPTION="Spreadsheet Calculator Improvised -- An ncurses spreadsheet program for terminal"
+HOMEPAGE="https://github.com/andmarti1424/sc-im"
+SRC_URI="https://github.com/andmarti1424/sc-im/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+S="${WORKDIR}/${P}/src"
+
+LICENSE="BSD-4"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~arm64 ~x86"
+IUSE="X plots wayland xls xlsx lua ods tmux"
```
[the remark this answers is not in the document]
Comment (ephemer4l) [reply app-office/sc-im/sc-im-0.8.3.ebuild:18]: Sorry, what should I be ordering here? Thank you so much for the review by the way!

**Comment by ephemer4l** [comment]
> With `+X` it fails to
> 
> ```
> checking for X11/Xmu/Atoms.h... no
> configure: error: *** X11/Xmu/Atoms.h is missing ***
> ```

I can't reproduce this. When enabling the X flag xclip gets culled in with libXmu and I see no error. How can I go about reproducing it?

File: app-office/sc-im/sc-im-0.8.3.ebuild (hunk 2 above)
[the remark this answers is not in the document]
Comment (juippis) [reply app-office/sc-im/sc-im-0.8.3.ebuild:18]: The use flags in an alphabetical/lexicographical order please.

**Comment by juippis** [comment]
> > With `+X` it fails to
> > ```
> > checking for X11/Xmu/Atoms.h... no
> > configure: error: *** X11/Xmu/Atoms.h is missing ***
> > ```
> 
> I can't reproduce this. When enabling the X flag xclip gets culled in with libXmu and I see no error. How can I go about reproducing it?

You're most likely missing a dependency for `X?` condition. Check with **equery b** in your system what brings the *X11/Xmu/Atoms.h* file and add that to DEPEND.

File: app-office/sc-im/sc-im-0.8.3.ebuild (hunk 2 above)
[the remark this answers is not in the document]
Comment (ephemer4l) [reply app-office/sc-im/sc-im-0.8.3.ebuild]: Oh! For some reason I thought it was already sorted. Thank you!

**Comment by ephemer4l** [comment]
> > > With `+X` it fails to
> > > ```
> > > checking for X11/Xmu/Atoms.h... no
> > > configure: error: *** X11/Xmu/Atoms.h is missing ***
> > > ```
> > 
> > 
> > I can't reproduce this. When enabling the X flag xclip gets culled in with libXmu and I see no error. How can I go about reproducing it?
> 
> You're most likely missing a dependency for `X?` condition. Check with **equery b** in your system what brings the _X11/Xmu/Atoms.h_ file and add that to DEPEND.

The missing header is provided by `x11-libs/libXmu`. What I can't understand is how you're getting this error when `X?` pulls in `x11-misc/xclip` which pulls in `x11-libs/libXmu`. On my system nothing else has the libXmu dependency other than xclip and when I'm installing with `+X` everything compiles fine.

**Review by juippis — APPROVED** [review]
Indeed and the error was in xclip and not in sc-im. Fixed with 
4d0ca652ea02e620be71378fc23847edd8ab856f

Thanks!

**Pull request closed without merging**